File: app/src/main/java/com/example/news/data/retrofit/Articles.java
package com.example.news.data.retrofit;

import com.google.gson.annotations.SerializedName;

public class Articles {

    @SerializedName("articles")
    public Article[] article;

    public static class Article {

        @SerializedName("source")
        public Source source;

        @SerializedName("title")
        public String title;

        @SerializedName("url")
        public String url;

        @SerializedName("urlToImage")
        public String urlToImage;

        @SerializedName("publishedAt")
        public String publishedAt;

        @SerializedName("content")
        public String content;
    }

    public static class Source {

        @SerializedName("name")
        public String name;
    }
}

File: app/src/main/java/com/example/news/data/retrofite_source/Articles.kt
package com.example.news.data.retrofite_source

import com.google.gson.annotations.SerializedName

data class Articles(
    @SerializedName("status") val status: String,
    @SerializedName("sources") val sources: MutableList<Article>
)

data class Article(
    @SerializedName("id") val id: String,
    @SerializedName("name") val name: String,
    @SerializedName("url") val url: String,
    @SerializedName("category") val category: String,
    @SerializedName("country") val country: String
)
